content/reference/custom_interactions.md

```diff
@@ -0,0 +1,221 @@
+{{ $message := complexMessage "buttons" $button }}
+{{ sendMessage nil $message }}
+```
+
+This button triggers our duck RegEx custom command.
+
+Now, let's add some more buttons.
+
+{{% notice info %}}
+
+You cannot have two message components with the same Custom ID in one message.
+
+{{% /notice %}}
+
+{{% notice info %}}
+
+Buttons with the "link" style cannot have a Custom ID, and instead require a URL field.
+
+Link style buttons do not trigger *interactions*.
+
+{{% /notice %}}
+
+```go
+{{ $button1 := cbutton "label" "Duck One" "custom_id" "buttons-duck-alpha" "style" "success" }}
+{{ $button2 := cbutton "emoji" (sdict "name" "🦆") "custom_id" "buttons-duck-beta" "style" "danger" }}
+{{ $button3 := cbutton "label" "Duck Three" "emoji" (sdict "name" "🦆") "url" "https://yagpdb.xyz" "style" "link" }}
+{{ $message := complexMessage "buttons" (cslice $button1 $button2 $button3) }}
+{{ sendMessage nil $message }}
+```
+
+![Many buttons](many_buttons.png)
+
+At this stage we have three buttons. Both of the first two buttons will trigger our duck trigger custom command, but the
+third button will not trigger any custom command. Link buttons do not create *interactions*.
+
+We can differentiate between the two buttons with `.StrippedID`, which returns the `.CustomID` but with the trigger and
+everything before it (in our case, `buttons-duck`) stripped off.
+
+Let's add in a select menu now.
+
+```go
+{{ $button1 := cbutton "label" "Duck One" "custom_id" "buttons-duck-alpha" "style" "success" }}
+{{ $button2 := cbutton "emoji" (sdict "name" "🦆") "custom_id" "buttons-duck-beta" "style" "danger" }}
+{{ $button3 := cbutton "label" "Duck Three" "emoji" (sdict "name" "🦆") "url" "https://yagpdb.xyz" "style" "link" }}
+
+{{ $menu := cmenu
+  "type" "text"
+  "placeholder" "Choose a terrible thing"
+  "custom_id" "menus-duck-alpha"
+  "options" (cslice
+    (sdict "label" "Ducks" "value" "opt-1" "default" true)
+    (sdict "label" "Duck" "value" "opt-2" "emoji" (sdict "name" "🦆"))
+    (sdict "label" "Half a Duck" "value" "opt-3" "description" "Don't let the smaller amount fool you."))
+  "max_values" 3 }}
+
+{{ $message := complexMessage "buttons" (cslice $button1 $button2 $button3) "menus" $menu }}
+{{ sendMessage nil $message }}
+```
+
+![Buttons and menus](buttons_and_menus.png)
+
+We now have two buttons and a menu which are triggering our duck custom command. We used to branch with `.StrippedID`,
+but now since we have two components whose stripped IDs are `-alpha`, we'll need to branch with `.IsMenu` too. If our
+custom command was triggered by the use of this select menu, we could also see that `.MenuType` returns `"string"`,
+since this menu is a string select type menu.
+
+##### Ordering Message Components
+
+A message can have 5 rows of components.
+A row of components can have 5 buttons, **or** 1 menu.
+
+Let's say I wanted to play tic tac toe. If I just add 9 buttons into the same slice in my complex message, they'll just
```

Comment:
```suggestion
Let's say we want to play tic tac toe. If we just add 9 buttons into the same slice in my complex message, they'll just
```

I'm assuming this a cookbook style you're aiming for, "we" seems more appropriate.